In [1]:
# importing required packages. 
import pandas as pd
import datetime as dt


In [2]:
#reading in the datas required to prepare the data quality report to the variable df.
df=pd.read_csv('2018_historic_weather_3.csv')

In [3]:
#Shows a dataframe overview.
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8800 entries, 0 to 8799
Data columns (total 25 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   dt                   8800 non-null   int64  
 1   dt_iso               8800 non-null   object 
 2   timezone             8800 non-null   int64  
 3   city_name            8800 non-null   object 
 4   lat                  8800 non-null   float64
 5   lon                  8800 non-null   float64
 6   temp                 8800 non-null   float64
 7   feels_like           8800 non-null   float64
 8   temp_min             8800 non-null   float64
 9   temp_max             8800 non-null   float64
 10  pressure             8800 non-null   int64  
 11  sea_level            0 non-null      float64
 12  grnd_level           0 non-null      float64
 13  humidity             8800 non-null   int64  
 14  wind_speed           8800 non-null   float64
 15  wind_deg             8800 non-null   i

In [4]:
# Dropped due to null values or irrevelant to data required.
df=df.drop(['dt_iso','sea_level','grnd_level','timezone','rain_1h','rain_3h','snow_1h','snow_3h'], axis = 1)

dt_iso dropped as repeated information.
'sea_level','grnd_level','rain_1h','rain_3h','snow_1h','snow_3h' dropped due to lack of information.
timezone dropped as only one timezone in the dataframe.

In [5]:
#Each number corresponds to a different Weather description.
df.weather_id.unique()

array([520, 802, 803, 801, 500, 521, 501, 311, 800, 300, 310, 741, 701,
       804, 612, 620, 600, 615, 621, 601, 301, 711, 502], dtype=int64)

In [6]:
df.weather_main.unique()

array(['Rain', 'Clouds', 'Drizzle', 'Clear', 'Fog', 'Mist', 'Snow',
       'Smoke'], dtype=object)

In [7]:
#Used to see what weather_id matches to what weather description
df.loc[(df['weather_id'] == 502 )]



dt city_name        lat      lon   temp  feels_like  temp_min  \
8461  1545116400    Dublin  53.349805 -6.26031  10.65        3.56      10.0   

      temp_max  pressure  humidity  wind_speed  wind_deg  clouds_all  \
8461     11.67       996        81         9.3       200          75   

      weather_id weather_main   weather_description weather_icon  
8461         502         Rain  heavy intensity rain          10n

reference table for weather id's and weather descriptions.

|Weather Id|Weather main|Weather description|
|---|---|---|
300|	Drizzle	light| intensity drizzle|
301|	Drizzle|	drizzle
310|	Drizzle|	light intensity drizzle rain
311|	Drizzle|	rain and drizzle
500|	Rain|	light rain
501|	Rain|	moderate rain
502|	Rain|	heavy intensity rain	
520|	Rain|	light intensity shower rain
521|	Rain|	proximity shower rain
600|	Snow|	light snow	
601|    Snow|	snow
612|	Snow|	light shower sleet	
615|	Snow|	light rain and snow
620|	Snow|	light shower snow	
621|	Snow|	shower snow
701|	Mist|	mist
711|	Smoke|	smoke
741|	Fog|	fog
800|	Clear|	sky is clear
801|	Clouds|	few clouds
802|	Clouds|	scattered clouds
803|	Clouds|	broken clouds	
804|	Clouds|	overcast clouds



In [8]:
df.head()

dt city_name        lat      lon  temp  feels_like  temp_min  \
0  1514764800    Dublin  53.349805 -6.26031  4.15       -6.49      3.84   
1  1514768400    Dublin  53.349805 -6.26031  4.14       -5.79      3.65   
2  1514772000    Dublin  53.349805 -6.26031  4.61       -5.77      3.85   
3  1514775600    Dublin  53.349805 -6.26031  4.64       -5.73      4.00   
4  1514779200    Dublin  53.349805 -6.26031  5.04       -4.91      4.11   

   temp_max  pressure  humidity  wind_speed  wind_deg  clouds_all  weather_id  \
0      5.79       990        87       12.86       240          40         520   
1      5.86       991        87       11.83       240          75         520   
2      5.99       990        81       12.35       240          40         802   
3      6.14       990        81       12.35       240          40         802   
4      6.22       990        81       11.83       240          40         802   

  weather_main          weather_description weather_icon  
0         Rain  light intensity shower rain          09n  
1         Rain  light intensity shower rain          09n  
2       Clouds             scattered clouds          03n  
3       Clouds             scattered clouds          03n  
4       Clouds             scattered clouds          03n

In [9]:
#Converted dt column to date time
df['dt']=pd.to_datetime(df['dt'],unit='s')


In [10]:
#split datetime into date and time columns 
df['date'] = pd.to_datetime(df['dt']).dt.date
df['time'] = pd.to_datetime(df['dt']).dt.time

In [11]:
#Dropped as superfluous to our needs.
df=df.drop(['lat','lon','city_name', 'temp_min','temp_max','pressure','humidity','wind_deg','weather_icon'], axis = 1)

In [12]:
#Remaining dataframe columns and their types
df.dtypes

dt                     datetime64[ns]
temp                          float64
feels_like                    float64
wind_speed                    float64
clouds_all                      int64
weather_id                      int64
weather_main                   object
weather_description            object
date                           object
time                           object
dtype: object

In [13]:
#dataframe view.
df.head()

dt  temp  feels_like  wind_speed  clouds_all  weather_id  \
0 2018-01-01 00:00:00  4.15       -6.49       12.86          40         520   
1 2018-01-01 01:00:00  4.14       -5.79       11.83          75         520   
2 2018-01-01 02:00:00  4.61       -5.77       12.35          40         802   
3 2018-01-01 03:00:00  4.64       -5.73       12.35          40         802   
4 2018-01-01 04:00:00  5.04       -4.91       11.83          40         802   

  weather_main          weather_description        date      time  
0         Rain  light intensity shower rain  2018-01-01  00:00:00  
1         Rain  light intensity shower rain  2018-01-01  01:00:00  
2       Clouds             scattered clouds  2018-01-01  02:00:00  
3       Clouds             scattered clouds  2018-01-01  03:00:00  
4       Clouds             scattered clouds  2018-01-01  04:00:00

Dropped columns.
the latitude=53.349805	
the longitude= -6.26031

In [14]:
#df.to_csv("cleaned_2018_historic_weather_3.csv")

In [15]:
sun_df=pd.read_csv('2018_sunrise_sunset.csv')

In [16]:
sun_df.head()

Date Sunrise Sunset
0  01/01/2018   09:05  16:59
1  02/01/2018   09:05  17:00
2  03/01/2018   09:05  17:01
3  04/01/2018   09:04  17:02
4  05/01/2018   09:04  17:03